extensions/access_log_policy/plugin.cc

```diff
@@ -0,0 +1,137 @@
+/* Copyright 2019 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "extensions/access_log_policy/plugin.h"
+
+#include <google/protobuf/io/zero_copy_stream_impl_lite.h>
+
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_split.h"
+#include "extensions/common/istio_dimensions.h"
+#include "extensions/common/node_info.pb.h"
+#include "google/protobuf/util/json_util.h"
+
+#ifndef NULL_PLUGIN
+
+#include "base64.h"
+
+#else
+
+#include "common/common/base64.h"
+namespace Envoy {
+namespace Extensions {
+namespace Wasm {
+namespace AccessLogPolicy {
+namespace Plugin {
+using namespace ::Envoy::Extensions::Common::Wasm::Null::Plugin;
+using NullPluginRootRegistry =
+    ::Envoy::Extensions::Common::Wasm::Null::NullPluginRootRegistry;
+using google::protobuf::util::JsonParseOptions;
+using google::protobuf::util::Status;
+
+NULL_PLUGIN_ROOT_REGISTRY;
+
+#endif
+
+namespace {
+
+bool setFilterStateValue(bool log) {
+  auto r = setFilterStateStringValue(::Wasm::Common::kAccessLogPolicyKey,
+                                     log ? "yes" : "no");
+  if (r != WasmResult::Ok) {
+    logWarn(toString(r));
+    return false;
+  }
+  return true;
+}
+
+}  // namespace
+
+constexpr absl::Duration kDefaultLogWindowDurationNanoseconds =
+    absl::Hours(12);  // 12h
```

Comment:
@bianpengyuan mentioned that absl time does not work well in Wasm since it makes system calls.

Reply:
@kyessenov ., @bianpengyuan : what is the suggested time measurement to use instead?
std::chrono? Previously I was using long int the way stackdriver plugin does for edge reporting : https://github.com/istio/proxy/blob/master/extensions/stackdriver/stackdriver.h#L108:12...

Reply:
Both would work. Just don't depend on absl/time to make it run in Wasm.

Reply:
Yes long int would be ideal if you want to compile the filter to wasm.

Reply:
Thanks.. will change back to long int..
Looks like we are missing some test coverage wherein we should have an ability to be able to run filters within wasm easily..